### ISSUE TITLE
Support for 'bold' namespace in fonts

### ISSUE THREAD

**Issue by robertLichtnow** [callstack/react-native-paper#1449]
## Describe the feature

Support for `bold` namespace in the `fonts` namespace of the theme

## Motivation

The reasons is getting closer to what material design showcases as 'common weights' in here https://material.io/design/typography/understanding-typography.html

## Related Issues

No related issues

**Comment by satya164** [comment]
Can you clarify your use case?

**Comment by robertLichtnow** [comment]
I'm making a design system for web (wrapping material ui) and react native (wrapping react native paper) that supports theming and one of the premises of the theme is that the fonts have the `'light' | 'regular' | 'medium' | 'thin' | 'bold'` configurations.

**Comment by satya164** [comment]
You can put any additional stuff in the theme. I don't think it makes sense for us to add a `bold` to the default theme when we don't use it.

**Comment by joebochill** [comment]
@satya164 I have a similar use case to @robertLichtnow. We are writing a component library that builds off of react-native-paper. We have our own theme file (where we can add bold), but we would like our components to be compatible with the default theme (or custom themes) as well. If we specify bold and it's not there, it defaults to regular, which is less than ideal.

**Comment by joaovpmamede** [comment]
@satya164 I've added the custom fonts to the:
```
namespace ReactNativePaper {
    interface ThemeFonts {
      bold: ThemeFont;
    }
  }
}
```

but now the `theme` object passed to the `<PaperProvider theme={theme}>` is giving me:

```
Types of property 'fonts' are incompatible.
Type 'Fonts' is missing the following properties from type 'ThemeFonts': bold,
```

Any way of augmenting the `Fonts` type?

**Comment by likon416** [comment]
> You can put any additional stuff in the theme. I don't think it makes sense for us to add a `bold` to the default theme when we don't use it.

Doing a theme manager where you can't set bold because they dont use it. Wow, great idea.

**Comment by lghimfus** [comment]
> @satya164 I've added the custom fonts to the:
> 
> ```
> namespace ReactNativePaper {
>     interface ThemeFonts {
>       bold: ThemeFont;
>     }
>   }
> }
> ```
> 
> but now the `theme` object passed to the `<PaperProvider theme={theme}>` is giving me:
> 
> ```
> Types of property 'fonts' are incompatible.
> Type 'Fonts' is missing the following properties from type 'ThemeFonts': bold,
> ```
> 
> Any way of augmenting the `Fonts` type?

did you find a way to fix this?

**Comment by likon416** [comment]
@lghimfus I think I finally make it work today.

I'm using config like this:
android: {
    bold: {
      fontFamily: customTheme.fonts.boldFamily,
      fontWeight: 'bold',
    },

and later in styles I'm using fontWeight: 'bold' BUT you need to import Text from React Native not React Native paper library.

**Comment by joebochill** [comment]
Are you declaring the namespace changes globally?

```ts
declare global {
    namespace ReactNativePaper {
        /* Theme Overrides */
    }
}
```

**Comment by likon416** [comment]
@joebochill I do not. just using fontconfig like I mentioned above

**Comment by joebochill** [comment]
I see the same thing if I try to use the `DefaultTheme` from RNP. When you override the type definition for the Theme, technically the one provided by RNP no longer matches. You'll have to either 1) not use the default theme or 2) throw a @ts-ignore on it and make sure you have fallbacks wherever they are needed.

We managed to get our library to work with our custom theme (that includes a bold font and some additional color values) and the default theme by adding fallbacks to a default theme property anywhere that we used one of our custom additions, e.g.:

```tsx
fontFamily: theme.fonts.bold ? theme.fonts.bold.fontFamily : theme.fonts.medium.fontFamily
```
